HAR-4583 Ympäristöraportin korjaus

Kauko opened the pull request.

Kauko committed:
Älä näytä ympäristöraportilla poistettuja toteuma_materiaaleja

Kauko committed:
Hae-urakan-materiaalit kyselyn ei tarvi palauttaa kokonaismäärää

Hae-urakan-materiaalit kyselyä käytetään ainoastaan suunnittelutietojen
hakemiseen, eikä kyseisessä näkymässä käytetä kokonaismäärää.

Kauko committed:
Poista vanhentunut hae-ymparistoraportti-urakoittain

Kauko committed:
Poista vanha ympäristöraporttikysely

Kauko committed:
Käytä cache-taulua, näytä suunnittelut ympäristöraportissa

Ympäristöraporttiin on tehty kolme korjausta:
- Älä ota mukaan toteuma_materiaaleja, jotka on poistettu
- Käytä reittipiste_materiaalien hakuun cache-taulua
- Tee päivämäärävertailu suunnitteluun oikein, jotta tiedot palautuvat

Kauko committed:
Lisää raporteille :arvo-ja-yksikko solu

Kauko committed:
Näytä ympäristöraportilla yksiköt

Kauko committed:
Älä näytä yhteenlaskusolua jos materiaalia ei käytetty

Kauko committed:
Näytä teksti, että suolalle ei toimi toteuma%

Tästä on tullut niin monta palautetta, ja on priorisoitu
alas, että sama kertoa käyttäjille.

Kauko committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

Kauko committed:
Näytä myös suunnittelutiedolle yksikkö

Kauko committed:
Ota huomioon ympäristöraportin testissä raporttisolut

Kauko committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

Kauko committed:
deploy2.retry .gitignoreen

Kauko committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

tatut committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

tatut commented on src/clj/harja/palvelin/raportointi/raportit/ymparisto.clj:
```diff
@@ -88,6 +88,7 @@
 
     [:raportti {:nimi raportin-nimi
                 :orientaatio :landscape}
+     [:teksti "Raportti ei vielä kykene laskemaan Suolan toteumaprosentteja."]
```
Miksi tällainen pahoitteluviesti on raportissa? Sana "vielä" viittaisi, että se on kohta tai jossain vaiheessa tulossa. Onko niin?

Kauko replied:
Lisäsin viestin, koska tuosta tulee jatkuvasti palautetta. Käyttäjät varmaan luulevat, että sen PITÄISI toimia, mutta bugaa. Näinhän ei ole.

En kyl tiedä, milloin ominaisuus lisätään. Monestihan tuosta on taidettu asiakkaan kanssa puhua, ja aina puskettu prioriteettia alaspäin. Voisi varmaan muokata viestiä, että yksinkertaisesti "ei laske".

tatut commented on src/cljs/harja/ui/raportti.cljs:
```diff
@@ -41,6 +41,12 @@
+  [:span.arvo-ja-yksikko
+   [:span.arvo (if fmt (fmt arvo) arvo)]
+   [:span " "]
+   [:span.osuus (str yksikko)]])
```
Pitäisikö tuon olla :span.yksikko, eikä luokalla "osuus"

Kauko replied:
Pitäisi

tatut commented on src/clj/harja/palvelin/raportointi/raportit/ymparisto.sql:
```diff
@@ -1,192 +1,63 @@
-   AND (:hallintayksikko::integer IS NULL OR u.hallintayksikko = :hallintayksikko)
-   AND (:urakkatyyppi::urakkatyyppi IS NULL OR u.tyyppi = :urakkatyyppi::urakkatyyppi)
- GROUP BY u.id, u.nimi, mk.id, mk.nimi, date_trunc('month', t.alkanut), rp.talvihoitoluokka
+SELECT
```
Aiemmassa kyselyssä olevat kommentit eri UNIONillä liitettävistä SELECTeistä olisivat mielestäni edelleen tarpeen. Niin pitkä kysely, että hyvä laittaa vähän kommenttia.

Kauko committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

Kauko committed:
Muuta ympäristöraportin viestiä suolan käytöstä

Kauko committed:
arvo-ja-yksikko elementille class yksikko

Keskusteltiin asiasta, ja todettiin, että näille voipi
lisätä "etukäteen" classit, vaikka css:ssä toteutusta ei olisikaan.

Kauko committed:
Kommentoi ympäristöraportti.sql

Kauko committed:
Lisää kommentteja ymparistoraportti.sql

Kauko committed:
Käytä ympäristöraportissa sopimuksen_kaytetty_materiaali taulua

Kauko committed:
Lisää puuttuva sarakkeen nimi

Kauko committed:
Siirrä materiaalin suunnittelun luonti omaan sql:ään

Aiemmin materiaalin_kaytto tauluun laitettiin suolankäyttöä testidatassa.
Suolankäytön suunnitelmat eivät kuitenkaan kuulu tähän tauluun.
Nyt luodaan hiekoitushiekkaa.

Kauko committed:
Lisää testiapureita raportin solujen hakuun

Kauko committed:
Vertaa ympäristö-, suola-, ja materiaaliraportin tuloksia testissä

Näiden raporttien tulokset pitäisi vastata toisiaan. Materiaaliraportti
ei tällä hetkellä valitettavasti näytä suunnittelun tuloksia, joten
vertailu ei ole aivan kattava.

Kauko committed:
Testaa, että ympäristöraportissa on jokainen materiaali vain kerran

Kauko committed:
Poista materiaalin duplikaattirivit ympäristöraportista

Varsinaista tulosta täydennetään kaikilla materiaalien nimillä, jotta
jokaista materiaalia kohti on yksi rivi. Tulosten hakuun oli lisätty
yksikkö, mutta tätä ei oltu lisätty materiaalit hakevaan kyselyyn.
Materiaalien tietoja käytetään group-by:ssa yhdistämään nämä
kaksi hakutulosta, ja koska tulokset olivat erit, ei yhdistäminen
onnistunut.

Kauko committed:
Korjaa testi, ympäristöraportti ei palauta prosenttia merkkijonona

tatut committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

Kauko committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

Kauko committed:
Merge branch 'HAR-4102' into HAR-4583_ymparistoraportti

Kauko committed:
Lisää puuttuva group by ympäristöraportille

Kauko committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

jarnovayrynen committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4583_ymparistoraportti

Kauko committed:
Yhdistä raportin tietojen haku urakoittain ja ei-urakoittain

Kauko committed:
Migraatio: hoitoluokattomat materiaalin käytöt nolla-luokkaan

Taulun luonnin yhteydessä ei oltu otettu huomioon, että reittipisteissä
on jo pisteitä, joille ei ole laskettu hoitoluokkaa. Uusille hoitoluokattomille
pisteille luokka asetetaan nollaksi, mutta näille vanhoille ei.

jarnovayrynen committed:
Rename V1_502__.sql to V1_503__.sql

jarnovayrynen committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

jarnovayrynen committed:
Merge branch 'HAR-4583_ymparistoraportti' of github.com:finnishtransportagency/harja into HAR-4583_ymparistoraportti

jarnovayrynen committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

jarnovayrynen committed:
Formatoi myös tuntematon talvihoitoluokka

jarnovayrynen committed:
Merge remote-tracking branch 'origin/HAR-4583_ymparistoraportti' into HAR-4583_ymparistoraportti

Kauko committed:
Merge branch 'develop' into HAR-4583_ymparistoraportti

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4583_ymparistoraportti

jarnovayrynen committed:
HAR-4232 Näytä ymp.raportissa talvisuolan max-määrä ja tot.prosentti

Näytä ympäristöraportissa talvisuolan maksimimäärä, käytetty määrä ja tot.prosentti.
Korjaa talvihoitoluokan fmt-muutoksesta rikkoontunut yksikkötesti.

jarnovayrynen committed:
Merge branch 'HAR-4583_ymparistoraportti' of github.com:finnishtransportagency/harja into HAR-4583_ymparistoraportti

jarnovayrynen committed:
Siisti talvisuolan maksimimäärän formatointia ymp.raportissa

jarnovayrynen committed:
Korjaa ymp.raportin urakoittain erittelyn päättely

jarnovayrynen committed:
Paranna talvisuolan maxmäärän näyttöä ja lisää testi

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4583_ymparistoraportti
